packages/ogmios/src/CardanoNode/OgmiosCardanoNode.ts

```diff
@@ -1,96 +1,57 @@
+import { BehaviorSubject, firstValueFrom, takeUntil } from 'rxjs';
 import {
-  Cardano,
   CardanoNode,
   EraSummary,
   GeneralCardanoNodeError,
   GeneralCardanoNodeErrorCode,
   HealthCheckResponse,
   StakeDistribution
 } from '@cardano-sdk/core';
-import {
-  ConnectionConfig,
-  LedgerStateQuery,
-  createConnectionObject,
-  createLedgerStateQueryClient,
-  getServerHealth
-} from '@cardano-ogmios/client';
 import { Logger } from 'ts-log';
-import { RunnableModule, contextLogger } from '@cardano-sdk/util';
-import { createInteractionContextWithLogger, ogmiosServerHealthToHealthCheckResponse } from '../util';
-import { queryEraSummaries, withCoreCardanoNodeError } from './queries';
+import { OgmiosObservableCardanoNode } from './OgmiosObservableCardanoNode';
+import { RunnableModule } from '@cardano-sdk/util';
 
 /**
  * Access cardano-node APIs via Ogmios
  *
  * @class OgmiosCardanoNode
  */
 export class OgmiosCardanoNode extends RunnableModule implements CardanoNode {
-  #stateQueryClient: LedgerStateQuery.LedgerStateQueryClient;
-  #logger: Logger;
-  #connectionConfig: ConnectionConfig;
+  #ogmiosObservableCardanoNode: OgmiosObservableCardanoNode;
+  #cancelAll$: BehaviorSubject<void>;
 
-  constructor(connectionConfig: ConnectionConfig, logger: Logger) {
+  constructor(ogmiosObservableCardanoNode: OgmiosObservableCardanoNode, logger: Logger) {
     super('OgmiosCardanoNode', logger);
-    this.#logger = contextLogger(logger, 'OgmiosCardanoNode');
-    this.#connectionConfig = connectionConfig;
+    this.#ogmiosObservableCardanoNode = ogmiosObservableCardanoNode;
+    this.#cancelAll$ = new BehaviorSubject<void>(undefined);
```

Comment:
BehaviorSubject always has a value (default supplied in the constructor) and emits immediately upon subscription. In this case immediately cancelling the subscription with `takeUntil` - probably throws `EmptyError`
```suggestion
    this.#cancelAll$ = new Subject<void>();
```

Reply:
Done here: https://github.com/input-output-hk/cardano-js-sdk/pull/1355/commits/0a42f6ac6602984634612286822dd3bfaf6e3195#diff-f39a864229b1800d755f14614cd8496fe47244069e7b95d5ae851156616cf34bR58